src/gtclang/Frontend/GTClangPreprocessorAction.cpp

```diff
@@ -406,7 +481,21 @@ class GTClangLexer {
           // Replace the '{' with '(){' to fix the missing argument-list of the Do-Method
           registerReplacement(token_.getLocation(), token_.getLocation(), "() {");
 
-          lexDoMethodBody(stencilKind, name, DoMethodLoc, storages);
+          lexDoMethodBody(stencilKind, name, DoMethodLoc, storages, storagesAllocatedOnTheFly);
+        }
+      }
+
+      if(DoWasLexed) {
+        // Add the found temporaries into the storage part of the Do-Method
+        if(!storagesAllocatedOnTheFly.empty()) {
+          registerReplacement(beforeFristDoMethod, beforeFristDoMethod, "void ");
```

Comment:
Shouldn't this be done regardless of whether you found some temporaries?

Reply:
we actually replace the void before the do-method with all the vars that are declared. If we did not find any, we can leave it